subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -590,10 +590,12 @@ int lwm2m_create_obj_inst(u16_t obj_id, u16_t obj_inst_id,
 	obj->instance_count++;
 	(*obj_inst)->obj = obj;
 	(*obj_inst)->obj_inst_id = obj_inst_id;
-	sprintf((*obj_inst)->path, "%u/%u", obj_id, obj_inst_id);
+	snprintf((*obj_inst)->path, MAX_RESOURCE_LEN, "%u/%u",
+		 obj_id, obj_inst_id);
 	for (i = 0; i < (*obj_inst)->resource_count; i++) {
-		sprintf((*obj_inst)->resources[i].path, "%u/%u/%u",
-			obj_id, obj_inst_id, (*obj_inst)->resources[i].res_id);
+		snprintf((*obj_inst)->resources[i].path, MAX_RESOURCE_LEN,
```

Comment:
`MAX_RESOURCE_LEN` isn't sufficient for this, unless `obj_id` and `obj_inst_id` can be clamped: it needs at least 33 bytes if the terminating NUL is included.  (10 characters per each unsigned integer, 2 for each slash, plus the terminating NUL.)

Reply:
Actually these are all u16_t types.  Which is 5 chars each (15 total) + 2 slashes and a NULL == 18.

Reply:
Ah, my bad, should have checked.